qiskit/opflow/gradients/circuit_qfis/lin_comb_full.py

```diff
@@ -73,20 +84,22 @@ def convert(
         elif isinstance(params, ParameterVector):
             params = params[:]  # unroll to list
 
-        # First, the operators are computed which can compensate for a potential phase-mismatch
-        # between target and trained state, i.e.〈ψ|∂lψ〉
-        gradient_states = LinComb()._gradient_states(
-            operator,
-            meas_op=phase_fix_observable,
-            target_params=params,
-            open_ctrl=False,
-            trim_after_grad_gate=True,
-        )
-        # if type(gradient_states) in [ListOp, SummedOp]:  # pylint: disable=unidiomatic-typecheck
-        if type(gradient_states) == ListOp:
-            phase_fix_states = gradient_states.oplist
-        else:
-            phase_fix_states = [gradient_states]
+        if phase_fix:
+            # First, the operators are computed which can compensate for a potential phase-mismatch
+            # between target and trained state, i.e.〈ψ|∂lψ〉
+            gradient_states = LinComb()._gradient_states(
+                operator,
+                meas_op=phase_fix_observable,
+                target_params=params,
+                open_ctrl=True,
+                trim_after_grad_gate=True,
+            )
+            # if type(gradient_states) in [ListOp, SummedOp]:  # pylint: disable=unidiomatic-typecheck
+
+            if type(gradient_states) == ListOp:
```

Comment:
Shall we use `isinstance`? Do we care about inheritance?

Reply:
I think we have to check explicitly for the `ListOp` type and not return `True` for inherited types. We have that in several cases in opflow, since eg a `ComposedOp` inherits from `ListOp`